src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -369,21 +418,84 @@ IEnumerable<WorkloadPackId> ExpandPacks (WorkloadId workloadId)
         public ISet<WorkloadInfo> GetWorkloadSuggestionForMissingPacks(IList<string> packIds)
         {
             var requestedPacks = new HashSet<WorkloadPackId>(packIds.Select(p => new WorkloadPackId(p)));
-            var expandedWorkloads = _workloads.Select(w => (w.Key, new HashSet<WorkloadPackId>(GetPacksInWorkload(w.Value))));
+            var expandedWorkloads = _workloads.Select(w => (w.Key, new HashSet<WorkloadPackId>(GetPacksInWorkload(w.Value.workload))));
             var finder = new WorkloadSuggestionFinder(GetInstalledPacks(), requestedPacks, expandedWorkloads);
 
             return new HashSet<WorkloadInfo>
             (
-                finder.GetBestSuggestion().Workloads.Select(s => new WorkloadInfo(s.ToString(), _workloads[s].Description))
+                finder.GetBestSuggestion().Workloads.Select(s => new WorkloadInfo(s.ToString(), _workloads[s].workload.Description))
             );
         }
 
         /// <summary>
         /// Returns the list of workloads defined by the manifests on disk
         /// </summary>
-        public IEnumerable<WorkloadDefinition> GetAvaliableWorkloads()
+        public IEnumerable<WorkloadDefinition> GetAvailableWorkloads()
         {
-            return _workloads.Values;
+            foreach ((WorkloadId _, (WorkloadDefinition workload, WorkloadManifest _)) in _workloads)
+            {
+                yield return workload;
+            }
+        }
+
+        /// <summary>
+        /// Determines which of the installed workloads has updates available in the advertising manifests.
+        /// </summary>
+        /// <param name="advertisingManifestResolver">A resolver that composes the advertising manifests with the installed manifests that do not have corresponding advertising manifests</param>
+        /// <param name="existingWorkloads">The IDs of all of the installed workloads</param>
+        /// <returns></returns>
+        public IEnumerable<WorkloadId> GetUpdatedWorkloads(WorkloadResolver advertisingManifestResolver, IEnumerable<WorkloadId> installedWorkloads)
```

Comment:
That is a new logic? WorkloadManifestUpdater.CalculateManifestUpdates is doing similar

Reply:
This is to replace it. This one handles redirects and cross-manifest dependencies.